src/rpc/RPCHelpers.cpp

```diff
@@ -1329,4 +1332,51 @@ toJsonWithBinaryTx(data::TransactionAndMetadata const& txnPlusMeta, std::uint32_
     return obj;
 }
 
+std::expected<ripple::STArray, Status>
+fetchCredentials(
+    std::optional<boost::json::array> const& credID,
+    std::shared_ptr<BackendInterface> const& backend,
+    ripple::LedgerHeader const& info,
+    Context const& ctx
+)
+{
+    ripple::STArray authCreds;
+    if (credID.value().size() > ripple::maxCredentialsArraySize) {
+        return Error{Status{
+            RippledError::rpcINVALID_PARAMS, "Invalid field 'credentials', not an array of CredentialID(hash256)."
+        }};
+    }
+
+    for (auto const& elem : credID.value()) {
+        ASSERT(elem.is_string(), "should already be checked in validators.hpp that elem is a string.");
+
+        ripple::uint256 credHash;
+        ASSERT(
+            credHash.parseHex(boost::json::value_to<std::string>(elem)),
+            "should already be checked in validators.hpp that elem is a uint256 hex"
+        );
+
+        auto const credKeylet = ripple::keylet::credential(credHash).key;
+        auto const credLedgerObject = backend->fetchLedgerObject(credKeylet, info.seq, ctx.yield);
+        if (!credLedgerObject)
+            return Error{Status{RippledError::rpcBAD_CREDENTIALS, "credentials aren't accepted."}};
+
+        auto credIt = ripple::SerialIter{credLedgerObject->data(), credLedgerObject->size()};
+        auto sleCred = ripple::SLE{credIt, credKeylet};
+
+        if (!credLedgerObject || (sleCred.getType() != ripple::ltCREDENTIAL) ||
+            ((sleCred.getFieldU32(ripple::sfFlags) & ripple::lsfAccepted) == 0u))
+            return Error{Status{RippledError::rpcBAD_CREDENTIALS}};
+
+        if (credentials::checkExpired(sleCred, info))
+            return Error{Status{RippledError::rpcBAD_CREDENTIALS}};
+
+        auto credential = ripple::STObject::makeInnerObject(ripple::sfCredential);
+        credential.setAccountID(ripple::sfIssuer, sleCred.getAccountID(ripple::sfIssuer));
```

Comment:
I am thinking do we need to convert it to STObject?  in `createAuthCredentials` , you will depack it anyway?

Reply:
Mainly because rippled's side converted to a credential object with `issuer` and `credential_type` field; but yes you're right that it will get depacked anyways. Do u think we should just not convert then or leave it the same?